aptos-move/framework/aptos-stdlib/sources/cryptography/algebra.move

```diff
@@ -16,14 +48,154 @@ module aptos_std::algebra {
+    public fun neg<S>(x: &Element<S>): Element<S> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        Element<S> {
+            handle: neg_internal<S>(x.handle)
+        }
+    }
+
+    /// Compute `x + y` for elements `x` and `y` of structure `S`.
     public fun add<S>(x: &Element<S>, y: &Element<S>): Element<S> {
         abort_unless_cryptography_algebra_natives_enabled();
         Element<S> {
             handle: add_internal<S>(x.handle, y.handle)
         }
     }
 
+    /// Compute `x - y` for elements `x` and `y` of a structure `S`.
+    public fun sub<S>(x: &Element<S>, y: &Element<S>): Element<S> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        Element<S> {
+            handle: sub_internal<S>(x.handle, y.handle)
+        }
+    }
+
+    /// Compute `x * y` for elements `x` and `y` of a structure `S`.
+    public fun mul<S>(x: &Element<S>, y: &Element<S>): Element<S> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        Element<S> {
+            handle: mul_internal<S>(x.handle, y.handle)
+        }
+    }
+
+    /// Try computing `x / y` for elements `x` and `y` of a structure `S`.
+    /// Return none if y equals to `zero<S>()`.
+    public fun div<S>(x: &Element<S>, y: &Element<S>): Option<Element<S>> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        let (succ, handle) = div_internal<S>(x.handle, y.handle);
+        if (succ) {
+            some(Element<S> { handle })
+        } else {
+            none()
+        }
+    }
+
+    /// Compute `x^2` for an element `x` of a structure `S`. Faster and cheaper than `mul(x, x)`.
+    public fun sqr<S>(x: &Element<S>): Element<S> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        Element<S> {
+            handle: sqr_internal<S>(x.handle)
+        }
+    }
+
+    /// Try computing `x^(-1)` for an element `x` of a structure `S`.
+    /// Return none if `x` equals to `zero<S>()`.
+    public fun inv<S>(x: &Element<S>): Option<Element<S>> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        let (succeeded, handle) = inv_internal<S>(x.handle);
+        if (succeeded) {
+            let scalar = Element<S> { handle };
+            some(scalar)
+        } else {
+            none()
+        }
+    }
+
+    /// Compute `2*P` for an element `P` of a structure `S`. Faster and cheaper than `P + P`.
+    public fun double<S>(element_p: &Element<S>): Element<S> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        Element<S> {
+            handle: double_internal<S>(element_p.handle)
+        }
+    }
+
+    /// Compute `k[0]*P[0]+...+k[n-1]*P[n-1]`, where
+    /// `P[]` are `n` elements of group `G` represented by parameter `elements`, and
+    /// `k[]` are `n` elements of the scalarfield `S` of group `G` represented by parameter `scalars`.
+    ///
+    /// Abort with code 0x010000 if the sizes of `elements` and `scalars` do not match.
```

Comment:
Here, we want to document without constants (as we generally do):

```
Abort with code `std::error::invalid_argument(E_NON_EQUAL_LENGTHS)`
```
We'll need to define `E_NON_EQUAL_LENGTHS` to be a constant equal to 1 in this module.

Then, we need to update the constant `MOVE_ABORT_CODE_INPUT_VECTOR_SIZES_NOT_MATCHING` in `aptos-move/framework/src/natives/cryptography/algebra/mod.rs` to be `0x01_0001`.

We need to do the same for `MOVE_ABORT_CODE_NOT_IMPLEMENTED`:
 - define an `E_NOT_IMPLEMENTED` constant equal to 2.
 - update its constant in Rust to be `0x0C_0002`